toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1092,6 +1106,91 @@ func (g *PkgGraph) AddGoalNode(goalName string, packages []*pkgjson.PackageVer,
 		}
 	}
 
+	// Expand the goal node if requested
+	if extraLayers > 0 {
+		g.addGoalNodeLayers(goalNode, extraLayers)
+	}
+
+	return
+}
+
+// addGoalNodeLayers will expand a goal node by one layer. For example, if the goaled node is "foo" (ie the goal node
+// points to "foo") and extraLevels is 1, the goal node will now link to all nodes which depend on "foo" as well as
+// "foo" itself. A node is considered to depend on "foo" if it is a run node that has edges connecting it to "foo"
+// without any other run nodes in between.
+//
+//	i.e., if "foo_run" -> "foo_build" -> "<Some Meta Node>" -> "bar_run" -> "bar_build", and we are expanding from "bar"
+//	with layers=1, only "foo_run" will be added to the goal nodes since "foo_build" and "<Some Meta Node>" are not run nodes.
```

Comment:
Let's keep the example consistent with the description above. Right now we've changed to dependency order.
```suggestion
//	E.g., if "bar_run" -> "bar_build" -> "<Some Meta Node>" -> "foo_run" -> "foo_build", and we are expanding from "foo"
//	with layers=1, only "bar_run" will be added to the goal nodes since "bar_build" and "<Some Meta Node>" are not run nodes.
```